src/frontend/js/clusive-prefs-django-store.js

```diff
@@ -0,0 +1,188 @@
+                args: ['{arguments}.1', "{that}.messageQueue", "{that}.lastRequestTime", "{that}"]
+            }
+        },
+        invokers: {
+            get: {
+                args: ['{that}', '{arguments}.0', '{that}.options.storeConfig']
+            }
+        }
+    });
+
+    fluid.defaults('clusive.prefs.djangoStore.writable', {
+        gradeNames: ['fluid.dataSource.writable'],
+        listeners: {
+            'onWrite.impl': {
+                listener: 'clusive.prefs.djangoStore.setUserPreferences',
+                args: ["{arguments}.0", "{arguments}.1", "{that}.messageQueue"]
+            }
+        },
+        invokers: {
+            set: {
+                args: ['{that}', '{arguments}.0', '{arguments}.1', '{that}.options.storeConfig']
+            }
+        }
+    });
+
+    fluid.makeGradeLinkage('clusive.prefs.djangoStore.linkage', ['fluid.dataSource.writable', 'clusive.prefs.djangoStore'], 'clusive.prefs.djangoStore.writable');
+
+    clusive.prefs.djangoStore.getUserPreferences = function(directModel, messageQueue, lastRequestTime, that) {
+        console.debug('clusive.prefs.djangoStore.getUserPreferences', directModel, messageQueue);
+
+        var djangoStorePromise = fluid.promise();
+
+        var preferenceChangeMessages = [].concat(messageQueue.queue).filter(function (item) {
+            if(item.content.type === "PC") {
+                return true;
+            }
+        }) 
+        
+        if(preferenceChangeMessages.length > 0) {
+            // Check for any preference changes in the message queue; 
+            // if there are any, the latest is them, not what's 
+            // available from the server; return the last preference change 
+            // from the queue
+            var latestPreferences = preferenceChangeMessages.pop().content.preferences;
+            console.debug("Get user preferences from the outstanding message queue", latestPreferences);
+            djangoStorePromise.resolve({
+                preferences: latestPreferences
+            });            
+            
+        } else {
+            // Implement debounce here
+            var currentTime = new Date();
+            var timeDiff;
+            if(lastRequestTime) {
+                timeDiff = currentTime.getTime() - lastRequestTime.getTime();
+            }
+
+            var debounce = directModel.debounce;
+            
+            var getURL = directModel.getURL;
+            
+            // Use cached result if within debounce time
+            if(timeDiff < debounce && that.lastResponse) {
+                console.debug('Get user preferences from cache of last result');
+                djangoStorePromise.resolve({
+                    preferences: that.lastResponse
+                });
+            } else if(that.requestIsInFlight) {
+                djangoStorePromise.reject('Won\'t get user preferences from server, another GET request is currently in flight')                
```

Comment:
I am seeing this message in the console when trying this out.  Is that expected?
Not clear that it resolves the request at all in this case.